### PR TITLE
Add a build rule for the boilerplate unit test.

### PR DESCRIPTION
**Pull request by spxtr**
We recently added unit tests that just run whenever you run `hack/verify_boilerplate.sh`, which really isn't the right time to do that. This adds a build rule instead.

### CODE REVIEW

**Comment by spxtr** [comment]
https://github.com/kubernetes/kubernetes/pull/39342

@k8s-bot verify test this

File: hack/boilerplate/BUILD [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 
 exports_files(glob(["*.txt"]))
+
+py_test(
+    name = "boilerplate_test",
+    srcs = [
+        "boilerplate_test.py",
+        "boilerplate.py",
+    ],
+    data = glob(["*.txt", "test/*"]),
```
Comment (mikedanese) [inline hack/boilerplate/BUILD]: Doesn't this need all source code to run

Comment (spxtr) [inline hack/boilerplate/BUILD]: No, this is unit tests for the verify script itself.

**Pull request merged**